### PR TITLE
[blog][posts]: Fix links for initial implementation of link checker 

### PR DESCRIPTION
**Pull request by DarhkVoyd**
**GitHub Issue:** #295 

**Summary**:
This PR addresses the broken links in the website blog posts and fixes the format of relative path links to support the initial implementation of a link checker to avoid blocking further PRs due to failing checks.

**Do you think resolving this issue might require an [Architectural Decision Record (ADR)](https://github.com/json-schema-org/community/blob/main/CONTRIBUTING.md#key-architectural-decisions)? (significant or noteworthy)**

No

Fixes: #295

### CODE REVIEW

**Comment by DarhkVoyd** [comment]
@Relequestual Kindly review the changes.

**Comment by Relequestual** [comment]
@Julian I've requested your review on this as there are a few Bowtie related link removals which might be just needing an update. I'm not sure removal is the right move here. Your call on those.

File: pages/blog/posts/bowtie-intro.md [hunk 1]
```diff
@@ -45,7 +45,7 @@ Bowtie attempts to bring these ideas to JSON Schema.
 
 From the existing [list of JSON Schema implementations](https://json-schema.org/implementations.html#validators), Bowtie already [supports](https://github.com/orgs/bowtie-json-schema/packages) *12* implementations across *9* programming languages, allowing anyone to run any of these implementations and see what they have to say about schemas and instances.
 
-It ships with a [command line program](https://bowtie-json-schema.readthedocs.io/en/latest/cli/), but perhaps more excitingly, ongoing automated runs of this CLI have been set up, such that Bowtie emits a [report across all of its supported implementations](https://bowtie-json-schema.github.io/bowtie/draft2020).
+It ships with a [command line program](https://bowtie-json-schema.readthedocs.io/en/latest/cli/), but perhaps more excitingly, ongoing automated runs of this CLI have been set up, such that Bowtie emits a report across all of its supported implementations.
```
Comment (Julian) [inline pages/blog/posts/bowtie-intro.md:48]:
```suggestion
It ships with a [command line program](https://docs.bowtie.report/en/stable/cli/) but perhaps more excitingly, ongoing automated runs of this CLI have been set up, such that Bowtie emits a [report across all of its supported implementations](https://bowtie.report).
```

File: pages/blog/posts/bowtie-intro.md [hunk 2]
```diff
@@ -146,7 +146,7 @@ This is probably the most beneficial area you can help in.
 I myself am not a frontend developer, in case that wasn't obvious.
 What I put together (with Bootstrap) is essentially the bare minimum of what's needed to show off results Bowtie emits.
 If you have more experience or headspace to think about how to effectively report test results, or compare implementations, please help!
-All of Bowtie's "frontend code" today lives in one place, [a Jinja2 template which gets formatted into a static single page site](https://github.com/bowtie-json-schema/bowtie/blob/main/bowtie/templates/report.html.j2).
```
Comment (Julian) [inline pages/blog/posts/bowtie-intro.md:149]: This isn't true anymore, but was true at the time of the post. Link-wise/stylistically I think the proper thing to do is probably just to link to a commit from the time, so that'd be e.g. https://github.com/bowtie-json-schema/bowtie/blob/b23644eb86cf9bcbdf1d98149cf3d976339f7cdf/bowtie/templates/report.html.j2

This sort of thing is why some blogs do "post updated on XYZ" things, so if we wanted to correct the *information* that seems like a decent thing to do but possibly relies on some infrastructure we may not have yet on the frontend (i.e. the ability to have multiple versions of the post, or at least to add that "updated on" note). I'd probably just change the link here for now and if we really cared we can come back later to make the note about accuracy because there undoubtedly are a bunch of things in this post which are now fixed/no longer true.

**Comment by Julian** [comment]
Thanks for the ping -- left some suggestions.

@DarhkVoyd minor feedback that hopefully is helpful is that when you submit a PR where you've made some *choices* (as for example here with some of the Bowtie links you chose to delete the link, presumably because you couldn't find an updated one, or thought it read better without one) you should call that out for the people reviewing the PR so that they know to pay extra attention, especially in this kind of PR where a lot of the changes are likely to be otherwise mechanical.

**Comment by DarhkVoyd** [comment]
@Julian Thank you so much for the feedback! Will surely keep that in mind going forwards. I have made the revision to just change the link for now.

File: pages/blog/posts/bowtie-intro.md [hunk 3]
```diff
@@ -146,8 +146,7 @@ This is probably the most beneficial area you can help in.
...
-Beyond generally "make it prettier, more responsive, or more usable", here are a few specific issues to look at:
+All of Bowtie's "frontend code" today lives in one place, [a Jinja2 template which gets formatted into a static single page site](https://github.com/bowtie-json-schema/bowtie/blob/b23644eb86cf9bcbdf1d98149cf3d976339f7cdf/bowtie/templates/report.html.j2).i
```
Comment (Julian) [inline pages/blog/posts/bowtie-intro.md:149]:
```suggestion
All of Bowtie's "frontend code" today lives in one place, [a Jinja2 template which gets formatted into a static single page site](https://github.com/bowtie-json-schema/bowtie/blob/b23644eb86cf9bcbdf1d98149cf3d976339f7cdf/bowtie/templates/report.html.j2).
```

File: pages/blog/posts/bowtie-intro.md (hunk 2 above)
Comment (DarhkVoyd) [reply pages/blog/posts/bowtie-intro.md:149]: @Julian For now I have just changed the link. I do not have deep understanding of the current infrastructure but we could just add another field to the markdown gray matter titled `last-updated` and just make desired changes, set the updated date and add this new updated date component somewhere next to the first release date.

File: pages/blog/posts/bowtie-intro.md (hunk 3 above)
Comment (DarhkVoyd) [reply pages/blog/posts/bowtie-intro.md]: Oops! There has been another mistake too! I have fixed both.

**Comment by DarhkVoyd** [comment]
@Julian Fixed my silly mistake!

File: pages/blog/posts/schema-static-analysis.md [hunk 4]
```diff
@@ -135,7 +135,7 @@ Dynamic references, on the other hand, generally boil down to one thing: dynamic
 
 ### Root schema dynamics
 
-Last year I wrote a [post](./dynamicref-and-generics) describing how to use `$dynamicRef` to model generic types in languages.  The idea went like this:
+Last year I wrote a [post](/blog/posts/dynamicref-and-generics) describing how to use `$dynamicRef` to model generic types in languages.  The idea went like this:
```
Comment (benjagm) [inline pages/blog/posts/schema-static-analysis.md:138]: Why not (dynamicref-and-generics) instead of (/blog/posts/dynamicref-and-generics)?

**Review by benjagm — CHANGES_REQUESTED** [review]
Thanks for the hard work on this!!

My only comment is regarding links from one blog to another blog in the same folder:
You are suggesting links adding "/blog/posts/" at the beginning of the url but this is the same folder so I think the name article will be enough. This will be a cleaner approach for these type of links.

Otherwise... great work!

File: pages/blog/posts/schema-static-analysis.md (hunk 4 above)
Comment (DarhkVoyd) [reply pages/blog/posts/schema-static-analysis.md:138]: Thanks for the positive feedback! 

Here's a quick overview of the link approach:

- At the time of this PR, the link checker we expected to use couldn't resolve paths in the repo context. As a workaround, I adopted the "/blog/posts/" convention.
- Currently, we're in the process of setting up a more robust link checker that will operate within the PR file context. 
- Both approaches have their merits, and we're flexible – either sticking with the current convention or opting for cleaner relative paths.

Let me know your preference.

File: pages/blog/posts/6-river-systems-case-study.md [hunk 5]
```diff
@@ -113,4 +113,4 @@ While 6 River Systems had previously used JSON Schema in a number of areas acros
 
 Thank you to [Jacob Thomas](https://www.linkedin.com/in/jake-thomas/) and 6 River Systems for enabling us to share this case study with you.
 
-Images used with permission.
\ No newline at end of file
+Images used with permission.
```
Comment (praveen-rikhari) [inline pages/blog/posts/6-river-systems-case-study.md:116]: @DarhkVoyd this is an unnecessary change right??

File: pages/blog/posts/and-then-there-were-three.md [hunk 6]
```diff
@@ -72,4 +72,4 @@ I'm really excited about the opportunity that Postman has given me to work full
 
 Thank you to everyone I've met on this journey, used my libraries, asked questions, contributed code, or just offered opinions!  I love being a part of this community!
 
-Cover photo by me 😁
\ No newline at end of file
+Cover photo by me 😁
```
Comment (praveen-rikhari) [inline pages/blog/posts/and-then-there-were-three.md:75]: same this shouldn't be in the commit

File: pages/blog/posts/applicability-json-schema-fundamentals-part-1.md [hunk 7]
```diff
@@ -429,6 +429,3 @@ All feedback is welcome. If you have questions or comments, you can find me on t
 - [JSON Schema live playground (ajv) for draft-07 with shareable links](https://jsonschema.dev)
 - [JSON Schema live playground (hyperjump validator) for all drafts (from draft-04)](https://json-schema.hyperjump.io)
 - [Links to RFC specification documents](/specification)
-
-
-Photo by [Heidi Fin](https://unsplash.com/@heidijfin?utm_source=unsplash&utm_medium=referral&utm_content=creditCopyText) on [Unsplash](https://unsplash.com/?utm_source=unsplash&utm_medium=referral&utm_content=creditCopyText)
```
Comment (praveen-rikhari) [inline pages/blog/posts/applicability-json-schema-fundamentals-part-1.md:434]: This was removed but nothing was added in its place.

File: pages/blog/posts/new-website.md [hunk 8]
```diff
@@ -29,7 +29,7 @@ The JSON Schema story started in 2007 when Chris Zyp proposed JSON Schema as a s
 
 Our new landing page is a modern, single destination with all the resources necessary for the different types of personas visiting the website: JSON Schema users, JSON Schema tooling implementers, and other organizations.
 
-The information architecture will make it easier for everyone to find the resources they need and make it easier for the Community to maintain and evolve our docs as the project grows. 
+The information architecture will make it easier for everyone to find the resources they need and make it easier for the Community to maintain and evolve our docs as the project grows.
```
Comment (praveen-rikhari) [inline pages/blog/posts/new-website.md:32]: why this change??

File: pages/blog/posts/bowtie-intro.md [hunk 9]
```diff
@@ -79,13 +79,13 @@ Bowtie is written in Python and published [on PyPI](https://pypi.org/project/bow
...
 ```sh
-$ pipx install bowtie-json-schema
+pipx install bowtie-json-schema
 ```
 
 which should give you a working Bowtie, which you can check via:
 
 ```sh
-$ bowtie --help
+bowtie --help
```
Comment (praveen-rikhari) [inline pages/blog/posts/bowtie-intro.md]: any specific reason we removed the $ here?? And is this related to this PR context??

File: pages/blog/posts/bowtie-intro.md [hunk 10]
```diff
@@ -79,13 +79,13 @@ Bowtie is written in Python and published [on PyPI](https://pypi.org/project/bow
 If you have no existing preferred setup for installing Python applications, [install `pipx` using the platform-specific instructions for your OS](https://pipx.pypa.io/stable/), then run:
 
 ```sh
-$ pipx install bowtie-json-schema
+pipx install bowtie-json-schema
```
Comment (praveen-rikhari) [inline pages/blog/posts/bowtie-intro.md]: why was the $ removed here??

**Review by praveen-rikhari — COMMENTED** [review]
Hey @DarhkVoyd did a quick scan for the PR, nice work with fixing all the broken links, thanks, i have commented a few things which are mostly related to unwanted changes, can we try to make the commits more atomic and if there was a change in a previous commit that you added in a later commit maybe squash them both together, Just a suggestion.

File: pages/blog/posts/and-then-there-were-three.md [hunk 11]
```diff
@@ -18,7 +18,7 @@ Hi, everybody!
 
 I'm Greg Dennis.  I'm the author of [`json-everything`](https://github.com/gregsdennis/json-everything), which aims to be an all-encompassing tool suite for JSON needs in .Net, including JSON Path, JSON Patch, and (of course) JSON Schema, amongst others.  I'm also responsible for driving and authoring the output requirements in the JSON Schema specification.
 
-Piggybacking on [Julian's news](hello-world-hello-postman), I'm also joining Postman to work on JSON Schema full time as a Specification & Tooling Architect!  This means that I will be working on updating the specification as well as helping define and implement validation and other JSON-Schema-adjacent tooling.  I also expect to be highly involved in the community by supporting other tooling implementors.
+Piggybacking on [Julian's news](/blog/posts/hello-world-hello-postman), I'm also joining Postman to work on JSON Schema full time as a Specification & Tooling Architect!  This means that I will be working on updating the specification as well as helping define and implement validation and other JSON-Schema-adjacent tooling.  I also expect to be highly involved in the community by supporting other tooling implementors.
```
Comment (benjagm) [inline pages/blog/posts/and-then-there-were-three.md:21]: For me the url here and the rest of the links from one blog to another should be just the name of the page
instead of (/blog/posts/hello-world-hello-postman)
it should be (hello-world-hello-postman)

All blog posts are in the shame folder.

File: pages/blog/posts/applicability-json-schema-fundamentals-part-1.md (hunk 7 above)
Comment (benjagm) [reply pages/blog/posts/applicability-json-schema-fundamentals-part-1.md:434]: Just leave this:

Photo by Heidi Fin on [Unsplash](https://unsplash.com/?utm_source=unsplash&utm_medium=referral&utm_content=creditCopyText).

File: pages/blog/posts/schema-static-analysis.md (hunk 4 above)
Comment (benjagm) [reply pages/blog/posts/schema-static-analysis.md:138]: The url here and the rest of the links from one blog to another should be just the name of the page because all are in the same folder.

File: pages/blog/posts/interpreting-output.md [hunk 12]
```diff
@@ -18,7 +18,7 @@ I've received a lot of questions (and purported bugs) and had quite a few discus
 
 Let's dig in.
 
-_The last time we [talked about output](./fixing-json-schema-output), it was to announce changes from the 2019-09/2020-12 version.  I'm going to use the new formats because it's easier to read and more compact._
+_The last time we [talked about output](/blog/posts/fixing-json-schema-output), it was to announce changes from the 2019-09/2020-12 version.  I'm going to use the new formats because it's easier to read and more compact._
```
Comment (benjagm) [inline pages/blog/posts/interpreting-output.md:21]: The url here and the rest of the links from one blog to another should be just the name of the page because all are in the same folder.

File: pages/blog/posts/custom-annotations-will-continue.md [hunk 13]
```diff
@@ -13,7 +13,7 @@ authors:
 excerpt: "Custom annotations are one of JSON Schemas most-used features.  Here's how we're still supporting them."
 ---
 
-[Last time](./the-last-breaking-change), I wrote about how we had to remove support for unknown keywords.  I ended that post with a note saying that we were still searching for a way that we can support non-functional custom keywords, that is, simple annotations.
+[Last time](/blog/posts/the-last-breaking-change), I wrote about how we had to remove support for unknown keywords.  I ended that post with a note saying that we were still searching for a way that we can support non-functional custom keywords, that is, simple annotations.
```
Comment (benjagm) [inline pages/blog/posts/custom-annotations-will-continue.md:16]: The url here and the rest of the links from one blog to another should be just the name of the page because all are in the same folder.

**Review by benjagm — CHANGES_REQUESTED** [review]
Please see my comments for links from one blog to another blog in the same folder.

Otherwise excellent job fixing this.

File: pages/blog/posts/and-then-there-were-three.md (hunk 11 above)
Comment (DarhkVoyd) [reply pages/blog/posts/and-then-there-were-three.md:21]: Understood!

File: pages/blog/posts/6-river-systems-case-study.md (hunk 5 above)
Comment (DarhkVoyd) [reply pages/blog/posts/6-river-systems-case-study.md:116]: Yeah, that's work of my markdown linter. Not sure if it changes anything.

File: pages/blog/posts/applicability-json-schema-fundamentals-part-1.md (hunk 7 above)
Comment (DarhkVoyd) [reply pages/blog/posts/applicability-json-schema-fundamentals-part-1.md:434]: Done!

**Comment by DarhkVoyd** [comment]
@benjagm I have incorporated the changes you suggested.

**Review by benjagm — APPROVED** [review]
Nice job with this! Thanks DV

**Review by Julian — APPROVED** [review]
